chat-plugins/uno.js

```diff
@@ -0,0 +1,563 @@
+			this.players[i].destroy();
+		}
+		delete this.room.game;
+	}
+}
+
+class UNOgamePlayer extends Rooms.RoomGamePlayer {
+	constructor(user, game) {
+		super(user, game);
+		this.hand = [];
+	}
+
+	canPlayWildFour() {
+		let colour = this.game.topCard.colour;
+
+		if (this.hand.some(c => c.colour === colour)) return false;
+		return true;
+	}
+
+	hasCard(cardName) {
+		let matched = this.hand.filter(c => c.name === cardName);
+		if (!matched.length) return false;
+		return matched[0]; // returns the first instance of the matched card
+	}
+
+	removeCard(cardName) {
+		for (let i = 0; i < this.hand.length; i++) {
+			if (this.hand[i].name === cardName) {
+				this.hand.splice(i, 1);
+				break;
+			}
+		}
+	}
+
+	buildHand() {
+		return this.hand.sort((a, b) => {
+			if (a.colour > b.colour) return 1;
+			if (a.colour < b.colour) return -1;
+			if (a.value > b.value) return 1;
+			return -1;
+		})
+		.map((c, i) => cardHTML(c, i === this.hand.length - 1));
+	}
+
+	sendDisplay() {
+		let hand = this.buildHand().join("");
+		let players = `<p><strong>Players (${this.game.playerCount}):</strong></p>` + this.game.getPlayers(true).join("<br />");
+		let draw = "<button style=\"border: 2px solid rgba(0 , 0 , 0 , 0.59); background-color: skyblue; border-radius: 8px; padding: 5px; font-weight: bold; width: 50%\" name=send value=\"/uno draw\">Draw a card!</button>";
+		let pass = "<button style=\"border: 2px solid rgba(0 , 0 , 0 , 0.59); background-color: pink; border-radius: 8px; padding: 5px; font-weight: bold; width: 50%\" name=send value=\"/uno pass\">Pass!</button>";
+
+		let top = `<strong>Top Card</strong>: <font color=${this.game.topCard.colour}>${this.game.topCard.name}</font>`;
+
+		// clear previous display and show new display
+		this.sendRoom("|uhtmlchange|uno-hand|");
+		this.sendRoom(`|uhtml|uno-hand|<table style="width: 100%; table-layout: fixed;">${this.game.currentPlayer === this.userid ? `<tr><td colspan=2></td><td style="border-radius: 50px 0 0 0; border: 1px solid; padding: 5px; text-align: right;">${top}</td></tr>` : ""}<tr><td colspan=2 style="border: 1px solid; padding: 5px;"><div style="overflow-x: auto; white-space: nowrap; width: 100%;">${hand}</div></td><td style="border: 1px solid; vertical-align: top; padding: 5px;"><div style="overflow-y: scroll;">${players}</div></td></tr>${this.game.currentPlayer === this.userid ? `<tr><td colspan=3 style="border: 1px solid;">${draw}${pass}</td></tr>` : ""}</table>`);
+	}
+}
+
+exports.commands = {
+	uno: {
+		// roomowner commands
+		disable: function (target, room, user) {
+			if (!this.can('gamemanagement', null, room)) return;
+			if (room.unoDisabled) {
+				return this.errorReply("UNO is already disabled.");
+			}
+			room.unoDisabled = true;
+			if (room.chatRoomData) {
+				room.chatRoomData.unoDisabled = true;
+				Rooms.global.writeChatRoomData();
+			}
+			return this.sendReply("UNO has been disabled for this room.");
+		},
+
+		enable: function (target, room, user) {
```

Comment:
mhm i'll PR this if this passes.

On this note, this follows the standard set in the hangman code. @Zarel would you want something more unified to control all minigame permissions in rooms?